gnssanalysis/filenames.py

```diff
@@ -722,6 +753,35 @@ def determine_name_props_from_filename(filename: str) -> Dict[str, Any]:
     }
 
 
+def warn_on_unexpected_filename(input_file: pathlib.Path) -> bool:
+    expected_file_name = determine_file_name(input_file, defaults={}, overrides={})
+    if input_file.name != expected_file_name:
+        logging.warning(f"File name: '{input_file.name}' "
+                       f"didn't match expected: '{expected_file_name}'. "
+                       "Contents may be incorrect and lead to failures."
+                       )
+        return False
```

Comment:
Just thinking about the name of the function `warn_on...`, should this return as `True` when it does find an issue? (i.e. when it should warn)

Reply:
I would concur and further suggest that warning is not really the responsibility of the library code. Detecting differences is probably in scope but warning is something that should be done in application code (like the ops comparison system).
Given that I'm wondering if this file should either just compare two names (at which point it's relatively simple and maybe could disappear?) or maybe tell you exactly which name properties differ between what's detected and what the current name is (if we can extract properties from the filename)?

Reply:
Good points. Detecting which bit differed is relatively easy given the properties dict...
Would you surface that information through an exception, if not a log message?

Reply:
I would just return the information in some format (maybe a possible empty list of items that disagree). This is library code and so in my head these tests are equivalent to specialised versions of `==` and you wouldn't ever expect a comparison operator to log or throw an exception, it's just making a comparison and then what to do about that comparison is really up to application code, in our case scripts in `ginan` or in the operations code.

Reply:
Restructured to provide a discrepancy checker function here in gnssanalysis. That is now leveraged by a wrapper function in the other code which logs errors in that context. Makes it more generic, and also more streamlined.

Reply:
Looks like it is **not** leveraged in other codebases at present. So for now the new and changed functionality appears unused.